Lib/extractor/formats/opentype.py

```diff
@@ -164,33 +171,24 @@ def _byteCodeToTtxAssembly(program):
 
 def _extractCompositeFlags(glyph, dest_glyph):
     # Find the lib key or add it
-    if (
-        "public.objectLibs" not in dest_glyph.lib
-        or "public.objectIdentifiers"
-        not in dest_glyph.lib["public.objectLibs"]
-    ):
-        dest_glyph.lib["public.objectLibs"] = {
-            "public.objectIdentifiers": {}
-        }
-    object_ids = dest_glyph.lib["public.objectLibs"][
-        "public.objectIdentifiers"
-    ]
+    if OBJECT_LIBS_KEY not in dest_glyph.lib:
+        dest_glyph.lib[OBJECT_LIBS_KEY] = {}
+    object_libs = dest_glyph.lib[OBJECT_LIBS_KEY]
 
     for ci, c in enumerate(glyph.components):
         flags = {}
-        if c.flags & ROUND_XY_TO_GRID:
-            flags["round"] = True
-        if c.flags & USE_MY_METRICS:
-            flags["useMyMetrics"] = True
-        if c.flags & OVERLAP_COMPOUND:
-            flags["overlap"] = True
+        flags[TRUETYPE_ROUND_KEY] = bool(c.flags & ROUND_XY_TO_GRID)
+        flags[TRUETYPE_METRICS_KEY] = bool(c.flags & USE_MY_METRICS)
```

Comment:
Would it be better to only define  the flags when their values are true?

Reply:
Hm, that would probably depend on how presence/absence of the flags is interpreted by authoring tools or at compile time. I probably was thinking along the lines of

- Flag is not present: Use heuristic or use authoring tool default value
- Flag is present: Use flag value

But I guess that is not specifed anywhere in the UFO spec ...?

So for anything TrueType-related I wanted to be on the safe side and always set the flags explicitly.

Reply:
That makes sense. Thanks.